apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -189,14 +199,25 @@ export default class LibraryPublisher extends React.Component {
     if (this.state.publishState === PublishState.ERROR_PUBLISH) {
       errorMessage = i18n.libraryPublishFail();
     }
+    if (this.state.publishState === PublishState.ERROR_UNPUBLISH) {
+      errorMessage = i18n.libraryUnPublishFail();
+    }
     return (
       <div>
         <p style={styles.alert}>{errorMessage}</p>
       </div>
     );
   };
 
+  unpublish = () => {
+    this.props.clientApi.delete(this.props.onUnpublishSuccess, error => {
+      console.warn(`Error publishing library: ${error}`);
+      this.setState({publishState: PublishState.ERROR_UNPUBLISH});
+    });
+  };
+
   render() {
+    let alreadyPublished = this.props.libraryDetails.alreadyPublished;
```

Comment:
Is this variable just for brevity over using `this.props.libraryDetails.alreadyPublished`?

Reply:
Short answer: Yes (with some other side benefits)

Long answer: It seems to be a new norm/best practice for accessing props/state in react - i.e. access all props/state at the beginning of your function and not later. There are a couple benefits. 
1. brevity
2. prevents you from giving a variable the same name as a state/prop variable (and therefore decreases confusion)
3. prevents you from attempting to change a props/state variable.
I did a quick google search to make sure I wasn't just making it up, and it looks like this pattern shows up in a couple articles and such:
https://towardsdatascience.com/react-best-practices-804def6d5215
https://brewhouse.io/blog/2015/03/24/best-practices-for-component-state-in-reactjs.html